src/server/services/pipeline-in-progress/processData.js

```diff
@@ -0,0 +1,169 @@
+				}
+				else if (startTimestamp.isBefore(prevReading.endTimestamp)) {
+					if (isCumulative) {
+						readingOK = false;
+						errMsg = 'The reading start time is before the previous endTime and cumulative so OED cannot use this reading.'
+					}
+					else if (!isFirst(prevReading.endTimestamp)) {
+						//Only treat this as a warning since the readings may be sent in a different order.
+						errMsg = 'The current reading startTime is not after the previous reading\'s end time. This is expected on the first reading ever.';
+					}
+				}
+			}
+			if (readingOK && Math.abs(startTimestamp.diff(prevReading.endTimestamp)) > Tgap) {
+				if (isCumulative) {
+					readingOK = false;
+					errMsg = 'The end of the previous reading is too far from the start of the next readings in cumulative data so drop the readings.';
+				}
+				else if (!isFirst(prevReading.endTimestamp)) {
+					//Only treat this as a warning since we expect a gap in the first ever reading.
+					errMsg = 'There is a gap in time between this reading and the previous reading.';
+				}
+			}
+			if (readingOK) {
+				//This reading can be used
+				if (isCumulative) {
+					meterReading1 = rows[index - 2 * readingRepetition][0]; // use this if regular
+					meterReading2 = rows[index - readingRepetition][0];
+					meterReading = meterReading2 - meterReading1; // use this ifCumulative
+				}
+				else {
+					meterReading = rows[index - readingRepetition][0];
+				}
+				// Reject negative readings
+				if (isCumulative && meterReading1 < 0) {
+					// If the value isCumulative and the first reading is negative reject unless cumulativeReset is true
+					log.error(`DETECTED A NEGATIVE VALUE WHILE HANDLING CUMULATIVE READINGS FROM METER ${meterID}, ` +
+						`ROW ${index - readingRepetition}. REJECTED ALL READINGS`);
+					return [];
+				}
+				// To handle cumulative readings that resets at midnight
+				if (meterReading < 0) {
+					// if meterReading is negative and cumulative check that the times fall within an acceptable reset range
+					if (handleCumulativeReset(cumulativeReset, resetStart, resetEnd, startTimestamp)) {
+						meterReading = meterReading2;
+					}
+					else {
+						//cumulativeReset is not expected so throw an error
+						errMsg = ('A negative meterReading has been detected but either cumulativeReset is not enabled, or the start time and end time of this reading is out of the reset range.')
+						readingOK = false;
+					}
+				}
+				if (readingOK) {
+					if (Math.abs(prevReading.endTimestamp.diff(prevReading.startTimestamp) - endTimestamp.diff(startTimestamp)) 
+						> Tlen && !isFirst(prevReading.endTimestamp)) {
+						errMsg = 'The previous reading has a different time length than the current reading.';
+					}
+					currentReading = new Reading(meterID, meterReading, startTimestamp, endTimestamp);
+					result.push(currentReading);
+				}
+			}
+			if (!readingOK) {
+				// An error occurred, for now log it, let the user know and continue
```

Comment:
Alright, I've added a loop to iterate over the reading numbers which were dropped. The idea is that in the file we might have 

HEADER ROW
READING VALUE # 1, START TIME, END TIME (READING #0)
READING VALUE # 2, START TIME, END TIME (READING #1)
... (READING #3)
and so on

and if we drop for example READING #0 because the data is cumulative and this is the first-ever data added to the db we let the user know:

Dropped Reading #0

and also the user will know ahead of time when parsing the data 

"Error Parsing Reading #0. Reading value gives READING VALUE # 1 with error message: The first reading must be dropped when dealing with cumulative data."

So the full log might be something like,

"Error Parsing Reading #0. Reading value gives READING VALUE # 1 with error message: The first reading must be dropped when dealing with cumulative data."

Dropped Reading #0

Now I don't know how this will work with other meter formats but I think it should be fine since ultimately each meter will contain readings and it should just be a matter of how we identify the readings which in my example is in canonical order of the CSV.